src/SqlAsyncCollector.cs

```diff
@@ -199,11 +198,11 @@ private async Task UpsertRowsAsync(IList<T> rows, SqlAttribute attribute, IConfi
                     }
                 }
 
-                if (tableInfo == null)
+                if (!(cachedTables[cacheKey] is TableInformation tableInfo))
                 {
                     TelemetryInstance.TrackEvent(TelemetryEventName.TableInfoCacheMiss, props);
                     // set the columnNames for supporting T as JObject since it doesn't have columns in the member info.
-                    tableInfo = await TableInformation.RetrieveTableInformationAsync(connection, fullTableName, this._logger, GetColumnNamesFromItem(rows.First()), this._serverProperties);
+                    tableInfo = TableInformation.RetrieveTableInformationAsync(connection, fullTableName, this._logger, GetColumnNamesFromItem(rows.First()), this._serverProperties);
```

Comment:
Should we also remove the Async from the method names? `RetrieveTableInformation`

Reply:
+1 - for any methods that no longer have await in them we should clean those up